apisix/core/resolver.lua

```diff
@@ -42,6 +58,21 @@ end
 -- @usage
 -- local ip, err = core.resolver.parse_domain("apache.org") -- "198.18.10.114"
 function _M.parse_domain(host)
+    local rev = HOSTS_IP_MATCH_CACHE[host]
+    if rev then
+        -- use ipv4 in high priority
+        local ip = rev["ipv4"]
+        if not ip then
+            ip = rev["ipv6"]
```

Comment:
We need to consider this setting:
https://github.com/apache/apisix/blob/2f4a4dba2c0cbc426ae99133c2546112ea71dba8/conf/config-default.yaml#L35

Reply:
I thought of this setting before, But I think when the default value of this setting is true , We also use ipv4 .  So  I think the  ip's mapping that in /etc/hosts decides, but I improve ipv4's priority

Reply:
When the users disable ipv6, we should not return the IPv6 address in /etc/hosts to them.